src/test/java/com/flightstats/hub/cluster/DistributedAsynchronousLockRunnerTest.java

```diff
@@ -0,0 +1,114 @@
+package com.flightstats.hub.cluster;
+
+import com.flightstats.hub.test.Integration;
+import org.apache.curator.framework.CuratorFramework;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class DistributedAsynchronousLockRunnerTest {
+    private static DistributedLeadershipLockManager lockManager;
+    private static AtomicReference<List<String>> lockList;
+
+    @BeforeClass
+    public static void setupCurator() throws Exception {
+        ZooKeeperState zooKeeperState = new ZooKeeperState();
+        CuratorFramework curator = Integration.startZooKeeper(zooKeeperState);
+        lockManager = new DistributedLeadershipLockManager(curator, zooKeeperState);
+    }
+
+    @Before
+    public void setup() {
+        lockList = new AtomicReference<>(newArrayList());
+    }
+
+    @Test
+    public void testDoesWorkPassedInFromLockable() throws Exception {
+        DistributedAsynchronousLockRunner distributedLockRunner = new DistributedAsynchronousLockRunner("/LockPath", lockManager);
+        CountDownLatch latch = new CountDownLatch(1);
+        SimpleTestLockable lockable = new SimpleTestLockable("lockable1", latch, 0);
+
+        Optional<LeadershipLock> runWithLock = distributedLockRunner.runWithLock(lockable, 1, TimeUnit.MILLISECONDS);
+
+        assertTrue(runWithLock.isPresent());
+
+        latch.await(1, TimeUnit.MINUTES);
+
+        assertEquals(newArrayList("lockable1"), lockList.get());
+    }
+
+    @Test
+    public void testPreventsTwoLockablesFromRunningAtTheSameTime() throws Exception {
```

Comment:
can we break name with underscores for more readability? something like like this https://www.baeldung.com/integration-testing-a-rest-api. I know underscores in name is not a java standard. But !

Reply:
In other project I have the habit of doing `test_methodName_assertionHere_givenWhatever` type names... so something like 'test_absoluteValue_returnsAPositiveNumber_givenANegativeNumber` I think keeping the "when"/method name near the start of the test name is helpful for scanning files and grouping, but I'm cool with whatever order of given/then feels good.